src/frontend/lsp/ocamlmerlin_lsp.ml

```diff
@@ -224,18 +224,28 @@ let on_request :
     begin match Query_commands.dispatch (Document.pipeline doc) command with
     | [] -> return (store, None)
     | (_loc, `Index _, _is_tail) :: _rest -> return (store, None)
-    | (_loc, `String contents, _is_tail) :: _rest ->
+    | (_loc, `String value, _is_tail) :: _rest ->
       let contents =
-        if
+        let supports_markdown =
           List.mem
             Lsp.Protocol.MarkupKind.Markdown
             client_capabilities.textDocument.hover.contentFormat
-        then "```\n" ^ contents ^ "\n```"
-        else contents
+        in
+        if supports_markdown
+        then {
+          Lsp.Protocol.MarkupContent.
+          value = "```ocaml\n" ^ value ^ "\n```";
```

Comment:
this is not critical, but String.concat or sprintf would be more performant

Reply:
Thanks, fixed.